### DISCUSSION TITLE
Trigger a required workflow on all PR's when applied on repo

### DISCUSSION THREAD

**Discussion by mjasion** [community/community#44516] [Actions]
### Select Topic Area

Product Feedback

### Body

Adding a required workflow to the repo forces us to push a commit to trigger it. Otherwise, it waits for the status with the note:
![image](https://user-images.githubusercontent.com/5058132/212757095-3d8c25e7-7810-4372-b0d7-143b9fc0cc32.png)

It would be nice when adding workflow would allow:
- triggering it on existing PRs - otherwise with "dismiss approvals with new commits" forces people to ask for approving it again,
- triggering a workflow manually - more like here https://github.com/orgs/community/discussions/43889

**Comment by MihirRajeshPanchal** [comment]
It is possible to trigger a workflow on an existing pull request (PR) or manually, but it requires additional configuration.

To trigger a workflow on an existing PR, you can use the on.pull_request_target event in your workflow file. This event triggers the workflow when a new commit is pushed to the target branch of a pull request.

You can also configure your workflow to trigger manually by using the workflow_dispatch event in your workflow file. This event allows you to manually trigger a workflow from the GitHub Actions page in your repository.

Additionally, you can use the /github/workflow/dispatch GitHub API endpoint to trigger the workflow manually.

Note that with manual triggering, you will need to include inputs in the API call, so that the action knows the values to use for any inputs defined in the action.

Also, for "dismiss approvals with new commits" you can set dismiss_stale_reviews: true in the workflow file. This will automatically dismiss any approvals that were given on a previous version of the code, and will require the PR to be re-approved when new commits are pushed.

By using above ways, you can trigger workflows on existing pull requests or manually.

**Reply by mjasion** [reply]
@MihirRajeshPanchal Your solution will work on normal workflows. My feedback relates to the [required workflows](https://docs.github.com/en/actions/using-workflows/required-workflows) feature, which was recently announced.

Before raising the feedback, I tested all of the ways you described. None of them worked.
